shell/src/main/java/alluxio/cli/docgen/ConfigurationDocGenerator.java

```diff
@@ -128,8 +145,18 @@ public static void writeCSVFile(Collection<? extends PropertyKey> defaultKeys,
     } finally {
       try {
         closer.close();
+        if (validate) {
+          compareFiles( CSV_FILE_NAMES,filePath);
+        }
       } catch (IOException e) {
         LOG.error("Error while flushing/closing Property Key CSV FileWriter", e);
+        for (String fileName : CSV_FILE_NAMES) {
```

Comment:
does this mean the temp files only get deleted if there's an error in closing the file? the temp files should be removed regardless of how close behaves so i think this block should be in the outer finally block

Reply:
oh now i see the delete file call in the compare files function. i think it's cleaner to have a single delete call since we can have both scenarios, with or without throwing an error, traverse through the finally block. otherwise it's easy to fall into the assumption i made where i did not think the deletion was handled properly